src/app/admin/user-panel/user-panel.component.html

```diff
@@ -0,0 +1,75 @@
+    <div class="admin-panel">
+      <h2 mat-dialog-title>Admin User Management</h2>
+
+      <mat-dialog-content>
+        <div class="table-container">
+          <table mat-table [dataSource]="dataSource" class="mat-elevation-z8">
+            <!-- Email Column -->
+            <ng-container matColumnDef="email">
+              <th mat-header-cell *matHeaderCellDef>Email</th>
+              <td mat-cell *matCellDef="let user">{{ user.email }}</td>
+            </ng-container>
+
+            <!-- Roles Column -->
+            <ng-container matColumnDef="roles">
+              <th mat-header-cell *matHeaderCellDef>Roles</th>
+              <td mat-cell *matCellDef="let user">
+                <div class="role-chips">
+                  <span class="role-chip" *ngFor="let role of user.roles">
+                    {{ role }}
+                  </span>
+                </div>
+              </td>
+            </ng-container>
+
+            <!-- Actions Column -->
+            <ng-container matColumnDef="actions">
+              <th mat-header-cell *matHeaderCellDef>Actions</th>
+              <td mat-cell *matCellDef="let user">
+                <button
+                  mat-icon-button
+                  [matMenuTriggerFor]="menu"
+                  aria-label="User actions"
+                >
+                  <mat-icon>more_vert</mat-icon>
+                </button>
+                <mat-menu #menu="matMenu">
```

Comment:
Since mat-menu is in every row of the table, it could be a performance issue.
The better solution would be to have one shared lazy menu definition (mat-menu outside the table) and pass data to it
https://material.angular.io/components/menu/overview#passing-in-data-to-a-menu

Reply:
Resolved - thanks for the tip